### ISSUE TITLE
Issue tracker label system for the development guidelines

### ISSUE THREAD

**Issue by rexim** [morganey-lang/organization#21]
Moved from https://github.com/morganey-lang/Morganey/issues/113
Split from #7

**Comment by ForNeVeR** [comment]
You could take a look how I've organized [Jabber-Net labels](https://github.com/ForNeVeR/Jabber-Net/labels).

**Comment by rexim** [comment]
@ForNeVeR yeah, I saw it, thanks. But I kinda disagree with it, because I consider things like `documentation` or `infrastructure` rather `topics` than `kinds`.

I define `kind` as a specific workflow for a ticket and also believe that the ticket can have only one `kind` (workflow) at the time.

For example, kind `incident` define workflow where we investigate the issue and constantly ask the reporter for clues. `bug` or `enhancement` defines a workflow where we prepare fix and submit it for the review.

I don't see `infrastructure` or `documentation` defining any workflow. They are topics. There can be such things as "bug of infrastructure" or "enhancement of documentation".

But that's just my opinion. Let's discuss that when I prepare a PR. :)

**Comment by ForNeVeR** [comment]
I personally think of these `kind`s as of responsibilities. That means we could have a guy who deals with bugs, another one who deals with docs, and one who implements features. But I can see your point.

**Comment by rexim** [comment]
That is interesting. What if we say that kinds define workflow and topics define responsibility?

**Comment by rexim** [comment]
So, those things become kinda orthogonal

**Comment by ForNeVeR** [comment]
That sounds interesting. Although I still don't know what's "workflow" you're talking about. From my point of view, "workflow" on each task is like "open → progress → closed". What type of workflow would you suggest for different kinds of issues?

**Comment by rexim** [comment]
> open → progress → closed

Our entire life has a similar "workflow", lol. "born → progress → die". 

By "workflow" I mean the details of the "progress" part. For examples I have my "split" issues. "split" is a kind. Nobody should "progress" directly on a split issue because its scope was completely delegated to its subtasks. The split issue is closed when all of its subtasks are closed.

**Comment by rexim** [comment]
Another interesting kind is "question". We progress on it by answering the question. Its workflow for example doesn't include a code review.
